templates/my.cnf.erb

```diff
@@ -1,21 +1,18 @@
 <% @options.sort.map do |k,v| -%>
-<% if v.is_a?(Hash) -%>
-[<%= k %>]
-<% @options[k].sort.map do |ki, vi| -%>
-<% if vi == true -%>
-<%= ki %>
-<% elsif vi.is_a?(Array) -%>
-<% vi.each do |vii| -%>
-<%= ki %> = <%= vii %>
-<% end -%>
-<% elsif vi and vi != '' -%>
-<%= ki %> = <%= vi %>
-<% elsif vi -%>
-<%= ki %>
-<% end -%>
-<% end -%>
-
-<% end -%>
+<%   if v.is_a?(Hash) -%>
+[<%=   k %>]
+<%     v.sort.map do |ki, vi| -%>
+<%       if vi == true or v == '' -%>
+<%=        ki %>
+<%       elsif vi.is_a?(Array) -%>
+<%         vi.each do |vii| -%>
+<%=          ki %> = <%= vii %>
+<%         end -%>
+<%       elsif vi != :undef -%>
+<%=        ki %> = <%= vi %>
```

Comment:
do i understand this code correct that  `foo = ''` will be set, but `foo = undef` will be unset?